appcues/src/main/java/com/appcues/ui/presentation/AppcuesViewModel.kt

```diff
@@ -25,15 +27,22 @@ import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.launch
 
+// TODO refactor dependency injection to reduce dependency count
 internal class AppcuesViewModel(
-    private val renderContext: RenderContext,
+    private val binding: PresentationBinding,
     private val coroutineScope: CoroutineScope,
     private val experienceRenderer: ExperienceRenderer,
+    private val analyticsTracker: AnalyticsTracker,
     private val actionProcessor: ActionProcessor,
-    private val onDismiss: () -> Unit,
-    val tapPassThroughHandler: (Offset) -> Unit,
 ) : ViewModel() {
 
+    interface PresentationBinding {
```

Comment:
got detekt warning of too many parameters on constructor... decided to group whats relevant to bind this viewModel to the presenter